java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java

```diff
@@ -54,6 +53,11 @@ public void visitNode(Tree tree) {
     if (Boolean.FALSE.equals(methodTree.isOverriding())) {
 
       for (VariableTree parameter : methodTree.parameters()) {
+        boolean hasRequestParamAnnotation = parameter.symbol().metadata().isAnnotatedWith("org.springframework.web.bind.annotation.RequestParam");
```

Comment:
According to [S6814](https://sonarsource.github.io/rspec/#/rspec/S6814), we should also support the `@PathVariable` annotation as it should also accept Optional-wrapped parameters